<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>viewMainWindow</class>
 <widget class="QMainWindow" name="viewMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">CSP</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="ViewHome" name="pageViewHome"/>
      <widget class="ViewConfigure" name="pageViewConfigure"/>
      <widget class="QWidget" name="pageEmpty"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuNew">
     <property name="title">
      <string>New Project</string>
     </property>
     <addaction name="actionNewChip"/>
    </widget>
    <addaction name="menuNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionImport"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="actionClose"/>
    <addaction name="actionReport"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionGenerate"/>
    <addaction name="actionPackageManager"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="menuProject"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidgetLeft">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>156</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetLeftContents">
    <layout class="QGridLayout" name="gridLayout">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QTreeView" name="treeView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetRight">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>58</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Property</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetRightContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="propertybrowser" name="treePropertyBrowser" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetBottomOutput">
   <property name="minimumSize">
    <size>
     <width>88</width>
     <height>107</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetBottomOutputContents">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="LogBox" name="LogBoxOutput">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetBottomConfigurations">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Configurations</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetBottomConfigurationsContents"/>
  </widget>
  <widget class="QDockWidget" name="dockWidgetBottomXmakeOutput">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Xmake Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetBottomXmakeOutputContents">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0">
      <widget class="LogBox" name="LogBoxXmakeOutput">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionGenerate"/>
   <addaction name="actionPackageManager"/>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>Load Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="text">
    <string>Import Project</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save Project As ...</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close Project</string>
   </property>
  </action>
  <action name="actionReport">
   <property name="text">
    <string>Generate Report</string>
   </property>
  </action>
  <action name="actionNewChip">
   <property name="text">
    <string>New Chip Project</string>
   </property>
  </action>
  <action name="actionGenerate">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icon/code-s-slash-line.svg</normaloff>:/icon/code-s-slash-line.svg</iconset>
   </property>
   <property name="text">
    <string>Generate code</string>
   </property>
  </action>
  <action name="actionPackageManager">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icon/dropbox-line.svg</normaloff>:/icon/dropbox-line.svg</iconset>
   </property>
   <property name="text">
    <string>Package Manager</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>propertybrowser</class>
   <extends>QWidget</extends>
   <header>propertybrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ViewConfigure</class>
   <extends>QWidget</extends>
   <header location="global">ViewConfigure.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ViewHome</class>
   <extends>QWidget</extends>
   <header location="global">ViewHome.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LogBox</class>
   <extends>QTextEdit</extends>
   <header>LogBox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
